cmd/commands/signature.go

```diff
@@ -48,68 +68,97 @@ func init() {
-
-  $ cpackget signature-create Vendor.Pack.1.2.3.pack
-
-By default the signature file will be created in the same directory as the provided pack.`,
+  $ cpackget signature-create Vendor.Pack.1.2.3.pack -k private.key -c certificate.pem`,
 	Args: cobra.ExactArgs(1),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		if signatureCreateCmdFlags.keyPath == "" && signatureCreateCmdFlags.passphrase != "" {
-			log.Error("-p/--passphrase is only specified when providing a key")
-			return errs.ErrIncorrectCmdArgs
+		// TODO: skipCertinfo and validation
+		if signatureCreateflags.keyPath == "" {
+			if !signatureCreateflags.certOnly || signatureCreateflags.pgp {
+				log.Error("specify private key file with the -k/-key flag")
+				return errs.ErrIncorrectCmdArgs
+			}
+		} else {
+			if signatureCreateflags.certOnly {
+				log.Error("-k/--key should not be provided in certificate-only mode")
+				return errs.ErrIncorrectCmdArgs
+			}
+		}
+		if signatureCreateflags.pgp {
+			if signatureCreateflags.certPath != "" {
+				log.Error("PGP signature scheme does not need a x509 certificate")
+				return errs.ErrIncorrectCmdArgs
+			}
+			if signatureCreateflags.skipCertValidation {
+				log.Error("PGP signature scheme does not validate certificates (--skip-validation)")
+				return errs.ErrIncorrectCmdArgs
+			}
+			if signatureCreateflags.skipInfo {
+				log.Error("PGP signature scheme does not display certificate info (--skip-info)")
+				return errs.ErrIncorrectCmdArgs
+			}
 		}
-		return cryptography.GenerateSignedChecksum(args[0], signatureCreateCmdFlags.keyPath, signatureCreateCmdFlags.outputDir, signatureCreateCmdFlags.passphrase, signatureCreateCmdFlags.outputB64)
+		return cryptography.SignPack(args[0], signatureCreateflags.certPath, signatureCreateflags.keyPath, signatureCreateflags.outputDir, Version, signatureCreateflags.certOnly, signatureCreateflags.skipCertValidation, signatureCreateflags.skipInfo)
 	},
 }
 
 var SignatureVerifyCmd = &cobra.Command{
-	Use:   "signature-verify [<local .checksum pack>] [<local private gpg key>]",
-	Short: "Verifies the integrity of a .checksum against its signature",
+	Use:   "signature-verify [<local .path pack>]",
+	Short: "Verifies a signed pack",
 	Long: `
-Verifies the integrity and authenticity of a .checksum file, by
-checking it against a provided .signature file (a detached PGP signature) and
-a private GPG key (either RSA or Curve25519).
+Verifies the integrity and authenticity of a pack signed
+with the "signature-create" command.
 
-The .signature and key files should have been created with the "signature-create" command,
-as they need to be in the PEM format.
+For more information on the signatures, use "cpackget help signature-create".
 
-If not specified by the -s/--sig-path flag, the .signature path will be read
-from the same directory as the .checksum file:
+If attempting to verify a PGP signed pack, use the -k/--pub-key flag to specify
+the publisher's public GPG key.
```

Comment:
I think we should decide whether we use the term PGP or GPG or is there a difference between a public GPG key vs. a public PGP key. Would we talk about an PGP signed pack if it is signed using a GPG private/public key pair?

Reply:
Yes, well caught. I will correct this, as PGP is the original standard, but since I've been using GnuPG to generate keys to develop, sometimes I get the two mixed up. Probably best to just use the term PGP, as GPG is an actual library/cli that may confuse users wanting to sign. GPG does output keys following the PGP standard.